src/main/kotlin/com/sourcegraph/cody/inspections/CodeActionQuickFix.kt

```diff
@@ -0,0 +1,106 @@
+package com.sourcegraph.cody.inspections
+
+import com.intellij.codeInsight.intention.IntentionAction
+import com.intellij.codeInsight.intention.PriorityAction
+import com.intellij.openapi.application.runInEdt
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiFile
+import com.sourcegraph.cody.agent.CodyAgentService
+import com.sourcegraph.cody.agent.protocol_generated.CodeActions_ProvideParams
+import com.sourcegraph.cody.agent.protocol_generated.CodeActions_TriggerParams
+import com.sourcegraph.cody.agent.protocol_generated.ProtocolLocation
+
+data class CodeActionQuickFixParams(
+    val title: String,
+    val kind: String?,
+    val location: ProtocolLocation,
+)
+
+class CodeActionQuickFix(private val params: CodeActionQuickFixParams) :
+    IntentionAction, PriorityAction {
+  companion object {
+    const val FAMILY_NAME = "Cody Code Action"
+  }
+
+  private val logger = Logger.getInstance(CodeActionQuickFix::class.java)
+
+  override fun getPriority(): PriorityAction.Priority {
+    return if (isFixAction()) {
+      PriorityAction.Priority.TOP
+    } else if (isExplainAction()) {
+      PriorityAction.Priority.LOW
+    } else {
+      PriorityAction.Priority.HIGH
+    }
+  }
+
+  override fun startInWriteAction(): Boolean {
+    // Commands will initialize their own edit sequence
+    return false
+  }
+
+  override fun getText(): String {
+    return params.title
+  }
+
+  private fun isFixAction(): Boolean {
+    // TODO: Ideally we have some flag indicating the semantic action type
+    return params.title.lowercase() == "ask cody to fix"
+  }
+
+  private fun isExplainAction(): Boolean {
+    // TODO: Ideally we have some flag indicating the semantic action type
+    return params.title.lowercase() == "ask cody to explain"
+  }
+
+  override fun getFamilyName(): String {
+    return FAMILY_NAME
+  }
+
+  override fun isAvailable(project: Project, editor: Editor?, file: PsiFile?): Boolean {
+    if (file == null || editor == null) {
+      return false
+    }
+
+    if (isExplainAction()) {
+      // TODO: Explain actions are temporarily disabled pending Protocol Support
+      return false
+    }
+
+    if (!(isFixAction() || isExplainAction())) {
+      // TODO: We temporarily disable unknown actions until we've verified they work.
```

Comment:
That two ifs translates to:
`isFixAction()` - `a`
`isExplainAction()` `b`

`!(a || b) && b`
`!((a || b) || !b)`
`!(a || b || !b)`
`!a`

So.. it is redundant indeed ;)
 Just `if (!(isFixAction() return false` will have the same effact as those two ifs